contracts/abstracted_account.algo.ts

```diff
@@ -1,5 +1,15 @@
 import { Contract } from '@algorandfoundation/tealscript';
 
+/**
+ * GateParams is a tuple of the parameters that are stored for each plugin,
+ * they dictate the accessibility of the plugin on the network and the duration
+ * of the plugin's access for a given address
+ */
+export interface GateParams {
```

Comment:
Why not just have a single `end` timestamp?

Reply:
Because i didn't think of it 🙃